darshan-util/darshan-logutils.h

```diff
@@ -273,4 +290,87 @@ int darshan_log_get_record(darshan_fd fd, int mod_idx, void **buf);
     memcpy(__ptr, __dst_char, 4); \
 } while(0)
 
+/*****************************************************************
+ * The functions in this section make up the accumulator API, which is a
+ * mechanism for aggregating records to produce derived metrics and
+ * summaries.
+ */
+
+/* opaque accumulator reference */
+struct darshan_accumulator_st;
+typedef struct darshan_accumulator_st* darshan_accumulator;
+
+/* Instantiate a stateful accumulator for a particular module type.
+ */
+int darshan_accumulator_create(darshan_module_id id,
+                               int64_t job_nprocs,
+                               darshan_accumulator*   new_accumulator);
+
+/* Add a record to the accumulator.  The record is an untyped void* (size
+ * implied by record type) following the convention of other logutils
+ * functions.  Multiple records may be injected at once by setting
+ * record_count > 1 and packing records into a contiguous memory region.
+ */
+int darshan_accumulator_inject(darshan_accumulator accumulator,
+                               void*               record_array,
+                               int                 record_count);
+
+struct darshan_file_category_counters {
+    int64_t count;                   /* number of files in this category */
+    int64_t total_read_volume_bytes; /* total read traffic volume */
+    int64_t total_write_volume_bytes;/* total write traffic volume */
+    int64_t max_read_volume_bytes;   /* maximum read traffic volume to 1 file */
+    int64_t max_write_volume_bytes;  /* maximum write traffic volume to 1 file */
+    int64_t total_max_offset_bytes;  /* summation of max_offsets */
+    int64_t max_offset_bytes;        /* largest max_offset */
```

Comment:
I guess the idea for having both totals and maxes for volumes/offsets is so that we can calculate the average and max volumes/offsets accessed for each file type? I think that makes sense, it was just notable to me, since I think the current incarnation of this "File count summary" table in darshan-job-summary.pl only provides one of those calculations (the table says "size", but I think it's using offset information). Maybe the next version of that table can summarize volume of I/O to different file types, in addition to offset information, since we're now capturing both.

Reply:
I'm not sure how useful this is.  The only reason I maintained it is to keep compatibility with what's reported by `darshan-parser` when using the `--file` option, but there were some bugs in the calculations there in the past.  The `total_max_offset_bytes` is a proxy for cumulative file size (regardless of how much was over-written or over-read per file) while the `max_offset_bytes` is a proxy for the size of the largest file (again, regardless of how much was over-written or over-read per file).  If they really were file sizes I think the utility would be obvious, but we have to use this proxy and the corresponding clunky names since we can't afford to actually stat the files and confirm their sizes.